greeter/README.md

```diff
@@ -1,21 +1,474 @@
+2. The greeting API can speak different languages based on the client locale. The support for new languages can be plugged in (extensibility and pluggability)
+3. Enable the caching with sweeping for REST requests (interception/observation)
+4. The greeting app and certain greeters have configuration options (configurability) _FIXME: we don't have this part, right?_
+
+![Greeting application diagram](https://raw.githubusercontent.com/strongloop/loopback-next/master/examples/greeting-app/greeting-app.png)
+
+This miniature project reflects many key aspects of a scalable and extensible applications, and relevant techniques will be covered in the workshop, like Inversion of Control, Dependency Injection, Decorators and Component.
+
+
+## Before We Begin
+To complete the steps in this tutorial, you need to install Node.js and the LoopBack 4 command line interface (CLI).
+
+1. Install Node.js version 8.9 or higher. 
+    See the [Node documentation](https://nodejs.org/en/download/) for installation instructions.
+    
+    Tip: We recommend to use [Node Version Manager (nvm)](https://github.com/nvm-sh/nvm) to make it easier to manage the Node.js versions you're using. 
+
+2. Install LoopBack 4 CLI
+
+    The LoopBack 4 CLI is a command-line interface that scaffolds an application as well as artifacts (for example, controllers, models, and repositories) for existing applications. The CLI provides the fastest and easiest way to get started with a LoopBack 4 project that adheres to best practices.
+
+    Install the CLI globally by running the following command:
+
+    ```sh
+    npm i -g @loopback/cli
+    ```
+
+## Part 1: Create a Greeting Application
+
+As the first step, we are going to create a greeting application that utilize the pre-existing LoopBack extension `greeter-extension`.
+
+### Step 1: Create a Message Type File
+We're going to create a `Message` type. This will be used as the response type that the GreetingService produces. In the `src` folder, create a file called `types.ts` with the following content:
+
+```ts
+/**
+ * Greeting message response
+ */
+export interface Message {
+  timestamp: Date;
+  language: string;
+  greeting: string;
+}
+```
+
+### Step 2: Create a Greeting Controller
+
+1. Create an endpoint `/greet/{name}` that takes a name and returns a greeting message in the language that is specified in the HTTP request. Run the Controller generator: 
+
+    ```sh
+    $ lb4 controller
+    ? Controller class name: Greeting
+    Controller Greeting will be created in src/controllers/greeting.controller.ts
+
+    ? What kind of controller would you like to generate? Empty Controller
+    create src/controllers/greeting.controller.ts
+    update src/controllers/index.ts
+
+    Controller Greeting was created in src/controllers/
+    ```
+
+2. In the generated GreetingController in `src/controllers/greeting.controller.ts`, modify the constructor:
```

Comment:
For this kind of code snippet, would be it better if we have
```ts
class GreetingController {
  constructor(
        @inject(GREETING_SERVICE) private greetingService: GreetingService,
        @inject(RestBindings.Http.REQUEST) private request: Request,
    ) {} 
  // define @post, @get here..
```
instead of just `constructor`?